### PR TITLE
Convert Java tests for JSDebuggerWebSocketClient to Kotlin

### PR DESCRIPTION
**Pull request by stewartsum**
## Summary:

Migrate Java tests to Kotlin for the JSDebuggerWebSocketClient class.

## Changelog:

Pick one each for the category and type tags:

[ANDROID] [CHANGED] - Update Java tests to Kotlin for the referenced file

## Test Plan:

All unit tests passed.

### CODE REVIEW

File: packages/react-native/ReactAndroid/src/test/java/com/facebook/react/devsupport/JSDebuggerWebSocketClientTest.kt [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+    "org.robolectric.*",
+    "androidx.*",
+    "android.*"
+)
+
+class JSDebuggerWebSocketClientTest {
+
+    @JvmField
+    @Rule
+    val rule = PowerMockRule()
```
Comment (cortinico) [inline packages/react-native/ReactAndroid/src/test/java/com/facebook/react/devsupport/JSDebuggerWebSocketClientTest.kt:39]:
```suggestion
    @get:Rule
    val rule = PowerMockRule()
```

File: packages/react-native/ReactAndroid/src/test/java/com/facebook/react/devsupport/JSDebuggerWebSocketClientTest.kt [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+    fun test_prepareJSRuntime_ShouldSendCorrectMessage() {
+        val cb = PowerMockito.mock(JSDebuggerCallback::class.java)
+        val client = PowerMockito.spy(JSDebuggerWebSocketClient())
+        client.prepareJSRuntime(cb)
+        PowerMockito.verifyPrivate(client)
+            .invoke("sendMessage", 0, "{\"id\":0,\"method\":\"prepareJSRuntime\"}")
+    }
+
+    @Test
+    @Throws(Exception::class)
```
Comment (cortinico) [inline packages/react-native/ReactAndroid/src/test/java/com/facebook/react/devsupport/JSDebuggerWebSocketClientTest.kt:52]: I think all those `@Throws(Exception::class)` can go, can you verify?

Comment (cortinico) [inline packages/react-native/ReactAndroid/src/test/java/com/facebook/react/devsupport/JSDebuggerWebSocketClientTest.kt:52]: I think all those `@Throws(Exception::class)` can go, can you verify?

File: packages/react-native/ReactAndroid/src/test/java/com/facebook/react/devsupport/JSDebuggerWebSocketClientTest.kt [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+    fun test_loadBundle_ShouldSendCorrectMessage() {
+        val cb = PowerMockito.mock(JSDebuggerCallback::class.java)
+        val client = PowerMockito.spy(JSDebuggerWebSocketClient())
+        val injectedObjects = HashMap<String, String>()
+        injectedObjects["key1"] = "value1"
+        injectedObjects["key2"] = "value2"
```
Comment (cortinico) [inline packages/react-native/ReactAndroid/src/test/java/com/facebook/react/devsupport/JSDebuggerWebSocketClientTest.kt:58]:
```suggestion
        val injectedObjects = mapOf("key1" to "value1", "key2" to "value2")
```

**Review by cortinico — CHANGES_REQUESTED** [review]
Thanks for sending this over @stewartsum Can I ask you to follow up on the suggestions I left?

File: packages/react-native/ReactAndroid/src/test/java/com/facebook/react/devsupport/JSDebuggerWebSocketClientTest.kt (hunk 2 above)
Comment (stewartsum) [reply packages/react-native/ReactAndroid/src/test/java/com/facebook/react/devsupport/JSDebuggerWebSocketClientTest.kt]: Removed all occurrences of `@Throws(Exception::class` and verified they were optional.

File: packages/react-native/ReactAndroid/src/test/java/com/facebook/react/devsupport/JSDebuggerWebSocketClientTest.kt (hunk 3 above)
Comment (cortinico) [reply packages/react-native/ReactAndroid/src/test/java/com/facebook/react/devsupport/JSDebuggerWebSocketClientTest.kt]: Yup that works 👍

**Review by cortinico — APPROVED** [review]
Thanks for sending this over @stewartsum 
I've rebased your PR and added a commit to remove PowerMock as we removed in `main` but your PR was opened against a older commit. We should be able to merge this now

**Pull request closed without merging**